tests/test/cycle.rs

```diff
@@ -163,9 +163,15 @@ fn overflow() {
         // Will try to prove S<G<Z>>: Q then S<G<G<Z>>>: Q etc ad infinitum
         goal {
             S<Z>: Q
-        } yields {
+        } yields[SolverChoice::slg(10, None)] {
             "Ambiguous; no inference guidance"
         }
+
+        goal {
+            S<Z>: Q
+        } yields[SolverChoice::recursive()] {
```

Comment:
This surprises me a bit.

Reply:
Yeah, I didn't fully investigate why it happens.

Reply:
This is actually fixed now that I had to make it flounder when truncation would happen.